compactor/src/parquet_file_filtering.rs

```diff
@@ -129,6 +129,103 @@ pub(crate) fn filter_parquet_files(
+    // Level 0 files sorted by max sequence number and level 1 files in arbitrary order for one
+    // partition
+    parquet_files_for_compaction: ParquetFilesForCompaction,
+    // Gauge for the number of Parquet file candidates
+    parquet_file_candidate_gauge: &Metric<U64Gauge>,
+    // Gauge for the number of bytes of Parquet file candidates
+    parquet_file_candidate_bytes_gauge: &Metric<U64Gauge>,
+) -> Vec<ParquetFile> {
+    let ParquetFilesForCompaction {
+        level_0,
+        level_1: mut remaining_level_1,
+    } = parquet_files_for_compaction;
+
+    if level_0.is_empty() {
+        info!("No level 0 files to consider for compaction");
+        return Vec::new();
+    }
+
+    // Guaranteed to exist because of the empty check and early return above. Also assuming all
+    // files are for the same partition.
+    let partition_id = level_0[0].partition_id;
+
+    let num_level_0_considering = level_0.len();
+    let num_level_1_considering = remaining_level_1.len();
+
+    // This will start by holding the level 1 files that are found to overlap an included level 0
+    // file. At the end of this function, the level 0 files are added to the end so they are sorted
+    // last.
+    let mut files_to_return = Vec::with_capacity(level_0.len() + remaining_level_1.len());
+    // Running total of the size, in bytes, of level 0 files and level 1 files for inclusion.
+    // The levels are counted up separately for metrics.
+    let mut total_level_0_bytes = 0;
+    let mut total_level_1_bytes = 0;
+
+    for level_0_file in &level_0 {
+        total_level_0_bytes += level_0_file.file_size_bytes as u64;
+
+        // Find all level 1 files that overlap with this level 0 file.
+        let (overlaps, non_overlaps): (Vec<_>, Vec<_>) = remaining_level_1
+            .into_iter()
+            .partition(|level_1_file| overlaps_in_time(level_1_file, level_0_file));
+
+        // Increase the running total by the size of all the overlapping level 1 files
+        total_level_1_bytes += overlaps.iter().map(|f| f.file_size_bytes).sum::<i64>() as u64;
+
+        // Move the overlapping level 1 files to `files_to_return` so they're not considered again;
+        // a level 1 file overlapping with one level 0 file is enough for its inclusion. This way,
+        // we also don't include level 1 files multiple times.
+        files_to_return.extend(overlaps);
+
+        // The remaining level 1 files to possibly include in future iterations are the remaining
+        // ones that did not overlap with this level 0 file.
+        remaining_level_1 = non_overlaps;
+    }
+
+    let num_level_0_compacting = level_0.len();
+    let num_level_1_compacting = files_to_return.len();
+
+    info!(
+        partition_id = partition_id.get(),
+        num_level_0_considering,
+        num_level_1_considering,
+        num_level_0_compacting,
+        num_level_1_compacting,
+        "filtered Parquet files for compaction",
```

Comment:
```suggestion
        "filtered cold Parquet files for compaction",
```

I think the `cold` will help. Otherwise, we will ask why compacting too many files